### PR TITLE
HW 1

### PR DESCRIPTION
**Pull request by limmark21**

### CODE REVIEW

**Review by PredelinaAsya — CHANGES_REQUESTED** [review]
Добрый день,

Спасибо за работу! Работа очень хорошая, ноутбук оформлен аккуратно, эксперименты проведены тщательно. Здорово, что комментируете получающиеся результаты и мотивируете свои действия. Есть несколько технических замечаний:

1. При разбиении датасета на train/val/test нужно было сделать стратификацию по целевой переменной (для этого используется аргумент `stratify` в `train_test_split`).
2. Вы пишите, что после замены `SGD` на `Adam` качество снижается. На самом деле это не так. Если посмотреть на `classification_report`, то станет ясно, что после обучения с `SGD` модель всегда предсказывает один класс — фактически, она не обучилась. После обучения с `Adam` она уже начинает обращать внимание на входные данные и предсказывает не тождественно 1. И по метрикам accuracy, и по f1 модель стала значительно лучше.
3. Очень здорово, что вы экспериментируете с архитектурой сети: шириной линейных слоёв и их количеством. Для такой простой архитектуры, как MLP, это, пожалуй, самые важные гиперпараметры. Могу лишь порекомендовать, что на практике лучше всего работают подходы, когда ширина резко возрастает на первом слое, а затем постепенно снижается. Например, можно было сделать что-то в духе: 15 (вход) -> 128 -> 32 -> 8 -> 1 (выход).
4. При проведении экспериментов нужно измерять качество на валидационной выборке, а не тестовой. Иначе, в ходе таких экспериментов происходит "подгонка под тест". В идеале метрики на тесте должны измерятся только один раз — в самом конце после всех экспериментов. Это позволит понять, насколько модель хороша на отложенной выборке, и проверить, не произошло ли переобучения под валидацию. **-1 балл**

Итого: **9 баллов** за домашнее задание. Отличный результат!
При исправлении замечаний выше можно повысить балл.

**Review by pacifikus — APPROVED** [review]
